src/sentry/search/events/datasets/spans_indexed.py

```diff
@@ -735,3 +761,154 @@ def _resolve_percentile_weighted(
             [args["column"], Function("toUInt64", [self.sampling_weight])],
             alias,
         )
+
+    def _query_total_counts(self) -> tuple[float | int, float | int]:
+        if self._cached_count is None:
+            total_query = spans_indexed.SpansEAPQueryBuilder(
+                dataset=self.builder.dataset,
+                params={},
+                snuba_params=self.builder.params,
+                selected_columns=["count()", "count_weighted()"],
+            )
+            total_results = total_query.run_query(Referrer.API_SPANS_TOTAL_COUNT_FIELD.value)
+            results = total_query.process_results(total_results)
+            if len(results["data"]) != 1:
+                raise Exception("Could not query population size")
+            self._cached_count = results["data"][0]["count"]
+            self._cached_count_weighted = results["data"][0]["count_weighted"]
+        return self._cached_count, self._cached_count_weighted
+
+    def _resolve_margin_of_error(
+        self,
+        args: Mapping[str, str | Column | SelectType | int | float],
+        alias: str | None = None,
+    ) -> SelectType:
+        """Calculates the Margin of error for a given value, but unfortunately basis the total count based on
+        extrapolated data"""
+        # both of these need to be aggregated without a query
+        total_samples, population_size = self._query_total_counts()
+        sampled_group = Function("count", [])
+        return Function(
+            "multiply",
+            [
+                # Based on a z score for a confidence level of 95%
+                1.96,
```

Comment:
We should probably pull this into a variable / option, so that the confidence level can be adjusted later without code changes. The correct z-score will probably depend on the experience of the estimate, which may of course change as this is worked on.

Reply:
:+1: Sure, I'll update that in a follow-up, I want to test this asap and based on @k-fish 's comments I want to do another version anyways